File: windows/src/CodexSwitch.Core/Configuration/Provider.cs
namespace CodexSwitch.Core.Configuration;

public enum Provider
{
    OpenAI,
    Sub2Api,
}

public static class ProviderExtensions
{
    public static string ToConfigValue(this Provider provider) => provider switch
    {
        Provider.OpenAI => "openai",
        Provider.Sub2Api => "sub2api",
        _ => throw new ArgumentOutOfRangeException(nameof(provider)),
    };
}

File: Sources/CodexProviderCore/Provider.swift
import Foundation

public enum Provider: String, CaseIterable, Equatable, Sendable {
    case openai
    case sub2api

    public var displayName: String {
        switch self {
        case .openai:
            return "OpenAI 官方"
        case .sub2api:
            return "Sub2API"
        }
    }
}
